core/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -196,36 +195,27 @@ private synchronized boolean connect() throws IOException {
         LOG.info("Trying to get local worker host : " + localHostName);
         workerNetAddress = mMasterClient.user_getWorker(false, localHostName);
         mIsLocal = true;
-      } catch (NoWorkerException e) {
+      } catch (Exception e) {
```

Comment:
That's one of the pains of Java, still need to do it...
On Jan 5, 2015 5:22 PM, "chang cheng" <notifications@github.com> wrote:

> In core/src/main/java/tachyon/worker/WorkerClient.java
> <https://github.com/amplab/tachyon/pull/602#discussion-diff-22503105>:
>
> > @@ -196,36 +195,27 @@ private synchronized boolean connect() throws IOException {
> >          LOG.info("Trying to get local worker host : " + localHostName);
> >          workerNetAddress = mMasterClient.user_getWorker(false, localHostName);
> >          mIsLocal = true;
> > -      } catch (NoWorkerException e) {
> > +      } catch (Exception e) {
>
> but all exceptions here have the same catch block... will be ugly then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amplab/tachyon/pull/602/files#r22503105>.
>